### PR TITLE
Upgrade handlebars.java to 4.0.7

### PR DESCRIPTION
**Pull request by ryuneeee**
Hello,
This project is nice, very comfortable use in Spring boot.

https://github.com/jknack/handlebars.java/blob/3b702a4a8ffeb07e52867fdf1ab69c9c414ae973/handlebars/src/main/java/com/github/jknack/handlebars/helper/ConditionalHelpers.java#L33

We can use conditional helpers of new feature in Handlebars.java 4.0.7

PR Included:
- Upgrade Handlebars.java to 4.0.7
- Change this project version to 0.3.1

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,24 +18,24 @@ repositories {
...
 dependencies {
-    compile 'pl.allegro.tech.boot:handlebars-spring-boot-starter:0.3.0'
+    compile 'pl.allegro.tech.boot:handlebars-spring-boot-starter:0.3.1'
 }
 ```
 
 ## Requirements
 
-Since version 0.3.0 handlebars-spring-boot-starter requires Spring Boot 2.0, Spring Framework 5 and Java 8.
+Since version 0.3.1 handlebars-spring-boot-starter requires Spring Boot 2.0, Spring Framework 5 and Java 8.
```
Comment (plesiecki) [inline README.md]: I guess this line should remain unchanged

Comment (ryuneeee) [reply README.md]: Of course!

**Comment by platan** [comment]
@ryuneeee thanks for your contribution!
A new version is released. I will let you know when the new version is available.

**Comment by platan** [comment]
@ryuneeee  handlebars-spring-boot-starter 0.3.1 is available http://repo1.maven.org/maven2/pl/allegro/tech/boot/handlebars-spring-boot-starter/0.3.1/

**Comment by ryuneeee** [comment]
@platan Thanks, platan!

**Pull request merged**